modules/revcontentBidAdapter.js

```diff
@@ -0,0 +1,290 @@
+        };
+      }
+    });
+
+    let data = {
+      id: bidderRequest.auctionId,
+      imp: imp,
+      site: {
+        id: widgetId,
+        domain: domain,
+        page: refererInfo,
+        cat: ['IAB17'],
+        publisher: {
+          id: userId,
+          domain: domain
+        }
+      },
+      device: {
+        ua: navigator.userAgent,
+        language: 'en'
+      },
+      user: {
+        id: 1
+      },
+      at: 2,
+      bcat: [
+        'IAB24',
+        'IAB25',
+        'IAB25-1',
+        'IAB25-2',
+        'IAB25-3',
+        'IAB25-4',
+        'IAB25-5',
+        'IAB25-6',
+        'IAB25-7',
+        'IAB26',
+        'IAB26-1',
+        'IAB26-2',
+        'IAB26-3',
+        'IAB26-4'
+      ]
+    };
+
+    serverRequests.push({
+      method: 'POST',
+      options: {
+        contentType: 'application/json'
+      },
+      url: endpoint,
+      data: JSON.stringify(data),
+      bid: validBidRequests
+    });
+
+    return serverRequests;
+  },
+  interpretResponse: function (serverResponse, originalBidRequest) {
+    if (!serverResponse.body) {
+      return;
+    }
+
+    const seatbid = serverResponse.body.seatbid[0];
+    const bidResponses = [];
+
+    for (var x in seatbid.bid) {
+      let adm = JSON.parse(seatbid.bid[x]['adm']);
+      let ad = {
+        clickUrl: adm.link.url
+      };
+
+      adm.assets.forEach(asset => {
+        switch (asset.id) {
+          case 3:
+            ad['image'] = {
+              url: asset.img.url,
+              height: 1,
+              width: 1
+            };
+            break;
+          case 0:
+            ad['title'] = asset.title.text;
+            break;
+          case 5:
+            ad['sponsoredBy'] = asset.data.value;
+            break;
+        }
+      });
+
+      var size = originalBidRequest.bid[0].params.size;
+
+      const bidResponse = {
+        bidder: BIDDER_CODE,
+        requestId: originalBidRequest.bid[0].bidId,
+        cpm: seatbid.bid[x]['price'],
+        creativeId: seatbid.bid[x]['adid'],
+        currency: 'USD',
+        netRevenue: true,
+        ttl: 360,
+        nurl: seatbid.bid[x]['nurl'],
+        bidderCode: 'revcontent',
+        mediaType: 'native',
+        native: ad,
+        width: size.width,
+        height: size.height,
+        ad: displayNative(ad, getTemplate(size, originalBidRequest.bid[0].params.template))
+      };
+
+      bidResponses.push(bidResponse);
+    }
+
+    return bidResponses;
+  },
+  onBidWon: function (bid) {
+    var winUrl = bid.nurl;
+    winUrl = winUrl.replace(/\$\{AUCTION_PRICE\}/, bid.cpm);
+    var host = extractHostname(winUrl);
+
+    ajax(winUrl + '&viewed=1', null, {withCredentials: true});
+    ajax('https://' + host + '/imp.php', null, 'v=' + encodeURIComponent(encodeURIComponent(getQueryVariable('d', winUrl))) + '&i=' + encodeURIComponent(window.location.href), {method: 'POST', contentType: 'application/x-www-form-urlencoded'});
```

Comment:
While we _do_ allow listening and firing imp trackers, can you explain the necessity to fire TWO imp trackers here?

We have allowed adapters to fire one in previous PR's, but wanted to know this use case and see if it can be updated to fire a single ajax call?

Reply:
Those are for different uses at the moment, we are working on reducing them into 1 call but to have accurate reporting we will need them both atm.

Reply:
@AzizSaleh Ok, so I spoke with the Prebid Org team and while we are a little hesitant as we want to limit things like this, we will allow it.

So whenever your team gets around to reducing it to 1 please make the corresponding change here!

It _may_ become a restricted thing in a future version of Prebid (Only allowing one track to fire per bidWon) but I will let you know ahead of time.